### PR TITLE
Cleanup dependencies

### PR DESCRIPTION
**Pull request by lgeiger**
This is the next round of monorepo cleanup. This PR removes extraneous and duplicated dependencies collected that made there way to the root `package.json`.

### CODE REVIEW

File: packages/connected-components/package.json [hunk 1]
```diff
@@ -33,17 +33,14 @@
     "@nteract/presentational-components": "^0.4.3",
     "@nteract/transforms": "^4.4.4",
     "babel-runtime": "^6.26.0",
-    "date-fns": "^1.29.0",
-    "react-dnd": "^5.0.0",
-    "react-dnd-html5-backend": "^5.0.0",
+    "rc-menu": "6.2.11",
```
Comment (lgeiger) [inline packages/connected-components/package.json:36]: This actually fixes a bug, when using `connected-components` outside the monorepo.

**Comment by lgeiger** [comment]
`react-virtualized`, `jsonschema`, `json-loader`, `ignore-styles`, `file-loader`, and `dom-helpers` are removed since they don't appear to be used in the monorepo.

**Comment by rgbkrk** [comment]
Wait, is our current version of next.js using this version of styled-jsx? I thought I cross checked that we were still on the styled-jsx 3.0.3 canary .

**Comment by rgbkrk** [comment]
I'm going to go ahead and merge, assume all is well in styled-jsx land and try it out.

**Pull request merged**